modules/management-console-agent/src/main/java/org/apache/ignite/agent/Agent.java

```diff
@@ -0,0 +1,451 @@
+import static org.apache.ignite.events.EventType.EVT_NODE_LEFT;
+import static org.apache.ignite.events.EventType.EVT_NODE_SEGMENTED;
+
+/**
+ * Management Agent.
+ */
+public class Agent extends ManagementConsoleProcessor {
+    /** Management Console configuration meta storage prefix. */
+    private static final String MANAGEMENT_CFG_META_STORAGE_PREFIX = "mgmt-console-cfg";
+
+    /** Discovery event on restart agent. */
+    private static final int[] EVTS_DISCOVERY = new int[] {EVT_NODE_FAILED, EVT_NODE_LEFT, EVT_NODE_SEGMENTED};
+
+    /** Not enabled events by default. */
+    private static final int[] NOT_ENABLED_EVTS = new int[] {
+        EVT_CLUSTER_ACTIVATED,
+        EVT_CLUSTER_DEACTIVATED,
+        EVT_CACHE_STARTED,
+        EVT_CACHE_STOPPED
+    };
+
+    /** Websocket manager. */
+    private WebSocketManager mgr;
+
+    /** Cluster service. */
+    private ClusterService clusterSrvc;
+
+    /** Tracing service. */
+    private TracingService tracingSrvc;
+
+    /** Span exporter. */
+    private ManagementConsoleSpanExporter spanExporter;
+
+    /** Events exporter. */
+    private EventsExporter evtsExporter;
+
+    /** Metric exporter. */
+    private MetricExporter metricExporter;
+
+    /** Metric service. */
+    private MetricsService metricSrvc;
+
+    /** Action service. */
+    private ActionService actSrvc;
+
+    /** Event service. */
+    private EventsService evtSrvc;
+
+    /** Node configuration service. */
+    private NodeConfigurationService nodeConfigurationSrvc;
+
+    /** Cache service. */
+    private CacheService cacheSrvc;
+
+    /** Execute service. */
+    private ThreadPoolExecutor connectPool;
+
+    /** Meta storage. */
+    private DistributedMetaStorage metaStorage;
+
+    /** Session registry. */
+    private SessionRegistry sessionRegistry;
+
+    /** Active server uri. */
+    private String curSrvUri;
+
+    /** If first connection error after successful connection. */
+    private AtomicBoolean disconnected = new AtomicBoolean();
+
+    /**
+     * @param ctx Kernal context.
+     */
+    public Agent(GridKernalContext ctx) {
+        super(ctx);
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onKernalStart(boolean active) {
+        metaStorage = ctx.distributedMetastorage();
+
+        evtsExporter = new EventsExporter(ctx);
+        spanExporter = new ManagementConsoleSpanExporter(ctx);
+        metricExporter = new MetricExporter(ctx);
+
+        // Connect to backend if local node is a coordinator or await coordinator change event.
+        if (isCoordinator(ctx.discovery().discoCache()))
+            connect();
+        else
+            ctx.event().addDiscoveryEventListener(this::launchAgentListener, EVTS_DISCOVERY);
+
+        evtsExporter.addLocalEventListener();
+        metricExporter.addMetricListener();
+
+        NodeConfigurationExporter exporter = new NodeConfigurationExporter(ctx);
+        exporter.export();
+        quiteStop(exporter);
```

Comment:
Why we stop this exporter on start?

Reply:
Removed. It's not needed anymore because we've removed CoordinatorSender.